3rdparty/p256-m/p256-m_driver_entrypoints.c

```diff
@@ -22,9 +22,9 @@
 #include "p256-m_driver_entrypoints.h"
 #include "p256-m/p256-m.h"
 #include "psa/crypto.h"
-#include "psa_crypto_driver_wrappers.h"
```

Comment:
This doesn't feel right: why can't p256m continue to include `psa_crypto_driver_wrappers.h` (which anyway includes `psa_crypto_driver_wrappers_no_static.h`), and continue to work? The fact that we have changed it so that some parts are static inline and some are not should not be visible to driver writers?

If I change this back, it breaks, so presumably any users with their own out-of-tree drivers will experience the same problem.

Reply:
It seems that the issue is related to `MBEDTLS_PRIVATE` - not sure how exactly

Reply:
@gilles-peskine-arm @tom-cosgrove-arm any thoughts on this?

Reply:
Whether to set `MBEDTLS_ALLOW_PRIVATE_ACCESS` needs to be decided before including any public header. Normally, in a library module, the first thing you do is include `"common.h"`, which takes care of defining `MBEDTLS_ALLOW_PRIVATE_ACCESS` before it includes any header that declares structures with private fields.

`psa_crypto_driver_wrappers.h` is an internal library header, and drivers are not supposed to include it. The fact that `p256-m_driver_entrypoints.c` includes it makes it not a normal driver, and instead part of the library. I'm not sure if that's ok and just makes it a bad example because it's integrated differently from third-party drivers, or if that's a genuine problem because it indicates that third-party drivers, which can't include `psa_crypto_driver_wrappers.h`, will be missing some critical definitions.

When driver support is finished, the driver wrapper generator will generate a public header containing the definitions that drivers need. We haven't started work on that part of driver support yet.

Reply:
Ah OK, in that case if we don't normally expect driver writers to include this header, then this isn't an issue.

Reply:
I think it's maybe a pre-existing issue in p256-m that it's using `psa_driver_wrapper_export_public_key` rather than `psa_export_public_key`, but that isn't anything to do with this PR.

Reply:
> I'm not sure if that's ok and just makes it a bad example because it's integrated differently from third-party drivers

The original reason to add P256-M to the code base was exactly as an example driver for third parties to use, so that makes this a problem.

> that isn't anything to do with this PR

Agreed; we'll have to worry about how to fix that after release (and then break any third-party drivers based on this example code we've provided)